src/audio/pipeline/pipeline-schedule.c

```diff
@@ -201,6 +202,25 @@ static enum task_state pipeline_task(void *arg)
 	 * copying below.
 	 */
 	err = pipeline_copy(p);
+	if (p->status == COMP_STATE_DRAINING) {
+		/* check the return value of the copy operation.
+		 *
+		 * if it's <= 0 then the draining opearation is over
+		 * and we need to send a notification to the host
+		 */
+		if (err <= 0) {
```

Comment:
what cases of `err > 0` are possible here and what should be done then? Do any `.copy` methods return `PPL_STATUS_PATH_STOP`? I haven't found any such cases

Reply:
For some reason I was under the wrong impression that `pipeline_copy` would return `>0` in case of success and `<=0` otherwise (=0 meaning it couldn't copy data) but that's not really the case.